dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/RequestFactory.kt

```diff
@@ -19,7 +19,9 @@ fun interface RequestFactory {
      * @param context Datadog SDK context.
      * @param batchData Raw data of the batch.
      * @param batchMetadata Raw metadata of the batch.
+     * @throws [Exception] in case the request could not be created.
      */
+    @kotlin.jvm.Throws
```

Comment:
I think this one is not needed. According to the [description](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-throws/) this is a compiler-related annotation which says to add `throws` keyword to the method signature in Java. This keyword will force the caller to handle this exception.

While in our case caller is not in Java, but in Kotlin, this obligation has no effect. Also not sure if `Throws` annotation without the list of exception will generate anything at all for JVM, will it?

Reply:
Probably mentioning it in the documentation will be enough. Not sure if `Throws` is copied to the implementation of the interface automatically by the IDE (so need to check interface description manually anyway in this case).